Plutarch/Maybe.hs

```diff
@@ -34,3 +36,10 @@ pfromJust = phoistAcyclic $
   plam $ \maybe -> pmatch maybe $ \case
     PNothing -> perror
     PJust a -> a
+
+-- | 'PMaybe' destructor equivalent to 'maybe'
+pmaybe :: Term (s :: S) (b :--> (a :--> b) :--> PMaybe a :--> b)
```

Comment:
This is likely not efficient.

Reply:
I'm not using it any more, so it can be removed if you find it dangerous.

Reply:
I think we should remove this unfortunately, yeah. Let's do this in Plutarch 2.0 efficiently.